api/src/main/java/com/epam/pipeline/manager/user/UserManager.java

```diff
@@ -95,26 +102,44 @@
     @Autowired
     private MetadataManager metadataManager;
 
+    @Autowired
+    private GrantPermissionManager permissionManager;
+
+    @Autowired
+    private DataStorageManager dataStorageManager;
+
+    @SuppressWarnings("PMD.AvoidCatchingGenericException")
     @Transactional(propagation = Propagation.REQUIRED)
     public PipelineUser createUser(String name, List<Long> roles,
                                    List<String> groups, Map<String, String> attributes,
                                    Long defaultStorageId) {
-        Assert.isTrue(StringUtils.isNotBlank(name),
-                messageHelper.getMessage(MessageConstants.ERROR_USER_NAME_REQUIRED));
-        String userName = name.trim().toUpperCase();
-        PipelineUser loadedUser = userDao.loadUserByName(userName);
-        Assert.isNull(loadedUser, messageHelper.getMessage(MessageConstants.ERROR_USER_NAME_EXISTS, name));
-        PipelineUser user = new PipelineUser(userName);
-        List<Long> userRoles = getNewUserRoles(roles);
-        user.setRoles(roleDao.loadRolesList(userRoles));
-        user.setGroups(groups);
-        user.setAttributes(attributes);
-        user.setDefaultStorageId(defaultStorageId);
-        storageValidator.validate(user);
-        log.info(messageHelper.getMessage(MessageConstants.INFO_CREATE_USER, userName));
-        return userDao.createUser(user, userRoles);
+        final PipelineUser newUser = createUser(name, roles, groups, attributes);
+        try {
+            return initUserDefaultStorage(defaultStorageId, newUser);
+        } catch (RuntimeException e) {
+            log.warn(messageHelper.getMessage(MessageConstants.ERROR_DEFAULT_STORAGE_CREATION, name, e.getMessage()));
+        }
+        return newUser;
     }
 
+    @Transactional(propagation = Propagation.REQUIRED)
+    public PipelineUser initUserDefaultStorage(final Long defaultStorageId, final PipelineUser newUser) {
+        final boolean shouldCreateDefaultHome =
+            preferenceManager.getPreference(SystemPreferences.DEFAULT_USER_DATA_STORAGE_ENABLED);
+        final Long storageId = Optional.ofNullable(defaultStorageId)
+            .filter(dataStorageManager::exists)
```

Comment:
This won't work as expected: if client specifies `defaultStorageId` and it doesn't exist a new storage will be create, I would expect that if invalid `defaultStorageId` is specified we will throw an error, as it worked previously.